apis/management.cattle.io/v3/alerting_types.go

```diff
@@ -0,0 +1,159 @@
+	TargetNode          TargetNode          `json:"targetNode,omitempty"`
+	TargetSystemService TargetSystemService `json:"targetSystemService,omitempty"`
+	TargetEvent         TargetEvent         `json:"targetEvent,omitempty"`
+}
+
+type ProjectAlertSpec struct {
+	AlertCommonSpec
+
+	ProjectName    string         `json:"projectName" norman:"type=reference[project]"`
+	TargetWorkload TargetWorkload `json:"targetWorkload,omitempty"`
+	TargetPod      TargetPod      `json:"targetPod,omitempty"`
+}
+
+type Recipient struct {
+	Recipient    string `json:"recipient,omitempty"`
+	NotifierID   string `json:"notifierId,omitempty" norman:"required,type=reference[notifier]"`
+	NotifierType string `json:"notifierType,omitempty" norman:"required,options=slack|email|pagerduty|webhook"`
+}
+
+type TargetNode struct {
+	ID           string            `json:"id,omitempty"`
+	Selector     map[string]string `json:"selector,omitempty"`
+	Condition    string            `json:"condition,omitempty" norman:"required,options=notready|mem|cpu,default=notready"`
+	MemThreshold int               `json:"memThreshold,omitempty" norman:"min=1,max=100"`
+	CPUThreshold int               `json:"cpuThreshold,omitempty" norman:"min=1"`
+}
+
+type TargetPod struct {
+	ID                     string `json:"id,omitempty" norman:"required"`
+	Condition              string `json:"condition,omitempty" norman:"required,options=notrunning|notscheduled|restarts,default=notrunning"`
+	RestartTimes           int    `json:"restartTimes,omitempty" norman:"min=1"`
+	RestartIntervalSeconds int    `json:"restartIntervalSeconds,omitempty"  norman:"min=1"`
+}
+
+type TargetEvent struct {
+	Type         string `json:"type,omitempty" norman:"required,options=Normal|Warning,default=warning"`
+	ResourceKind string `json:"resourceKind,omitempty" norman:"required,options=Pod|Node|Deployment|Statefulset|Daemonset"`
+}
+
+type TargetWorkload struct {
+	ID                    string            `json:"id,omitempty"`
+	Type                  string            `json:"type,omitempty" norman:"required,options=deployment|statefulset|daemonset,default=deployment"`
+	Selector              map[string]string `json:"selector,omitempty"`
+	UnavailablePercentage int               `json:"unavailablePercentage,omitempty" norman:"required,min=1,max=100"`
+}
+
+type TargetSystemService struct {
+	Condition string `json:"condition,omitempty" norman:"required,options=etcd|controller-manager|scheduler,default=scheduler"`
+}
+
+type AlertStatus struct {
+	State string `json:"state,omitempty" norman:"required,options=active|inactive|alerting|muted,default=active"`
+}
+
+type Notifier struct {
+	types.Namespaced
+
+	metav1.TypeMeta `json:",inline"`
+	// Standard object’s metadata. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec NotifierSpec `json:"spec"`
+	// Most recent observed status of the cluster. More info:
```

Comment:
Nit: comments seems to belong to cluster status

Reply:
Yep, I will remove the comment